qiskit/circuit/library/data_preparation/pauli_feature_map.py

```diff
@@ -116,7 +115,12 @@ def __init__(
         self,
         feature_dimension: Optional[int] = None,
         reps: int = 2,
-        entanglement: Union[str, List[List[int]], Callable[[int], List[int]]] = "full",
+        entanglement: Union[
+            str,
+            Dict[int, List[Tuple[int]]],
+            Dict[int, List[List[int]]],
+            Callable[[int], Union[str, Dict[int, List[Tuple[int]]], Dict[int, List[List[int]]]]],
```

Comment:
We only need one of `List[List]` or `List[Tuple]`, they are equivalent 🙂 
```suggestion
            Dict[int, List[Tuple[int]]],
            Callable[[int], Union[str, Dict[int, List[Tuple[int]]]]],
```